// Copyright 2022 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.cmrt.sdk.auto_scaling_service.v1;

option java_multiple_files = true;
option java_outer_classname = "AutoScalingServiceTestConfigurationKeysProto";
option java_package = "com.google.cmrt.sdk.auto_scaling_service.v1";

// For faster allocations of sub-messages.
option cc_enable_arenas = true;

// Configuratiion keys for testing.
enum TestClientConfigurationKeys {
  // Invalid key.
  UNKNOWN_TEST_AUTO_SCALING_CLIENT_CONFIGURATION_KEY = 0;
  // Optional. Only needed for integration test to override the cloud service
  // endpoint.
  CMRT_TEST_AUTO_SCALING_CLIENT_CLOUD_ENDPOINT_OVERRIDE = 1;
  // Required for GCP. It should be a real project in GCP which is
  // used for your test.
  CMRT_TEST_AUTO_SCALING_CLIENT_OWNER_ID = 2;
  // Optional for GCP. It should be a real zone which is used for your test. If
  // not set, use the default value us-central1-a.
  CMRT_TEST_AUTO_SCALING_CLIENT_ZONE = 3;
  // Optional for GCP. It should be a real instance id which is used for your
  // test. If not set, use the default value 12345678987654321.
  CMRT_TEST_AUTO_SCALING_CLIENT_INSTANCE_ID = 4;
  // Optional for GCP. The impersonate SA to talk to cloud storage. If not set,
  // will not use impersonate SA for testing.
  CMRT_TEST_GCP_AUTO_SCALING_CLIENT_IMPERSONATE_SERVICE_ACCOUNT = 5;
  // Optional for AWS. If not set, use the default value us-east-1.
  CMRT_TEST_AUTO_SCALING_CLIENT_REGION = 6;
}
